controllers/auth.js

```diff
@@ -0,0 +1,32 @@
+
+
+// controllers/authController.js
+const User = require('../models/User');
+const jwt = require('jsonwebtoken');
+
+// Register a new user
+exports.registerUser = async (req, res) => {
+    const { username, password } = req.body;
+    try {
+        const user = await User.create({ username, password });
```

Comment:
Your `User` model does not have a `username` field. It has the fiels `name`, `password` and `email`. And they are all required so your code won't work like this.